File: contracts/responses/DeviceResponses.ts
import { Device } from "../models/Device";

export interface DeviceResponse {
  device: Device;
}

export interface DeviceListResponse {
  devices: Device[];
}

export interface PairDeviceResponse {
  deviceId: string;
  isPaired: boolean;
  pairingStatus: string;
  device?: Device | null;
}

File: contracts/responses/DeviceResponses.kt
package contracts.responses

import contracts.models.Device
import kotlinx.serialization.Serializable

@Serializable
data class DeviceResponse(
    val device: Device
)

@Serializable
data class DeviceListResponse(
    val devices: List<Device> = emptyList()
)

@Serializable
data class PairDeviceResponse(
    val deviceId: String,
    val isPaired: Boolean,
    val pairingStatus: String,
    val device: Device? = null
)
